homeassistant/components/ihc/binary_sensor.py

```diff
@@ -67,12 +68,16 @@ def __init__(
         self.inverting = inverting
 
     @property
-    def device_class(self):
+    def device_class(self) -> BinarySensorDeviceClass | None:
         """Return the class of this sensor."""
-        return self._sensor_type
+        return (
+            BinarySensorDeviceClass(self._sensor_type)
+            if self._sensor_type in DEVICE_CLASSES
+            else None
+        )
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool | None:
```

Comment:
This is type hint improvement - unrelated to config-flow. Please move to preliminary/follow-up PR.

Reply:
Done in PR [90350](https://github.com/home-assistant/core/pull/90350)